addons/project_purchase/models/project_project.py

```diff
@@ -65,27 +65,24 @@ def action_open_project_purchase_orders(self):
 
     def action_profitability_items(self, section_name, domain=None, res_id=False):
         if section_name == 'purchase_order':
-            action = {
+            context = {'create': False, 'edit': False}
+            if res_id:
+                return {
+                    'name': self.env._('Purchase Orders'),
+                    'type': 'ir.actions.act_window',
+                    'res_model': 'purchase.order',
+                    'view_mode': 'form',
+                    'res_id': self.env['purchase.order'].search([('order_line', '=', res_id)]).id,
+                    'context': context,
+                }
+            return {
                 'name': self.env._('Purchase Order Items'),
                 'type': 'ir.actions.act_window',
                 'res_model': 'purchase.order.line',
                 'views': [[False, 'list'], [False, 'form']],
                 'domain': domain,
-                'context': {
-                    'create': False,
-                    'edit': False,
-                },
+                'context': context,
             }
-            if res_id:
-                action['res_id'] = res_id
```

Comment:
Why not simply updating line 79 and adding 2 other lines.

As no test have failed after your changes, that means this unit is not tested, so let's test it now to avoid breaking your work in the future.
```
                action['res_id'] = self.env['purchase.order'].search([('order_line', '=', res_id)]).id
                action['res_model'] = 'purchase.order'
                action['name'] = self.env._('Purchase Orders')
```

Reply:
> Why not simply updating line 79 and adding 2 other lines.
> 
> As no test have failed after your changes, that means this unit is not tested, so let's test it now to avoid breaking your work in the future.
> 
> ```
>                 action['res_id'] = self.env['purchase.order'].search([('order_line', '=', res_id)]).id
>                 action['res_model'] = 'purchase.order'
>                 action['name'] = self.env._('Purchase Orders')
> ```

Why should I check all the conditions if we only have a single purchase order?
so we have removed this code.
    
            `if 'views' in action:
                    action['views'] = [
                        (view_id, view_type)
                        for view_id, view_type in action['views']
                        if view_type == 'form'
                    ] or [False, 'form']
                action['view_mode'] = 'form'`

Reply:
I was just asking if you know the reason why it was done this way. so you did it to simplify this loop. okay noted thanks